k8s/logs-spack-io/deployments.yaml

```diff
@@ -0,0 +1,352 @@
+          valueFrom:
+            secretKeyRef:
+              name: logs-spack-io-secrets
+              key: mysql_database
+        - name: MATOMO_DATABASE_ADAPTER
+          value: mysql
+        - name: MATOMO_DATABASE_TABLES_PREFIX
+          valueFrom:
+            secretKeyRef:
+              name: logs-spack-io-secrets
+              key: matomo_database_tables_prefix
+        resources:
+          requests:
+            cpu: 1000m
+            memory: 8Gi
+          limits:
+            cpu: 1000m
+            memory: 8Gi
+        ports:
+        - name: http
+          containerPort: 9000
+        volumeMounts:
+        - mountPath: /usr/local/etc/php/conf.d/php-special.ini
+          name: config
+          subPath: php.ini
+        - mountPath: /var/www/html
+          name: html-files
+        - mountPath: /data
+          name: migration-logs
+      - image: mysql:8.0
+        name: mysql
+        env:
+        - name: MYSQL_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: logs-spack-io-secrets
+              key: mysql_root_password
+        - name: MYSQL_DATABASE
+          valueFrom:
+            secretKeyRef:
+              name: logs-spack-io-secrets
+              key: mysql_database
+        - name: MYSQL_USER
+          valueFrom:
+            secretKeyRef:
+              name: logs-spack-io-secrets
+              key: mysql_user
+        - name: MYSQL_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: logs-spack-io-secrets
+              key: mysql_password
+        ports:
+        - containerPort: 3306
+          name: mysql
+
+        # Important - this is recommended to be 500GB SSD - can we do that?
+        volumeMounts:
+        - name: matomo-db
+          mountPath: /var/lib/mysql
+        - name: matomo-configmap
+          mountPath: /etc/mysql/conf.d/more.cnf
+          subPath: more.cnf
+        livenessProbe: 
+          initialDelaySeconds: 10
+          timeoutSeconds: 10
+          periodSeconds: 30
+          failureThreshold: 5
+          tcpSocket:
+            port: mysql
+        readinessProbe: 
+          initialDelaySeconds: 20
+          timeoutSeconds: 10
+          periodSeconds: 30
+          failureThreshold: 5
+          tcpSocket:
+            port: mysql
+        resources:
+          limits:
+            cpu: "3.0"
+            memory: "3Gi"
+          requests:
+            cpu: "0.25"
+            memory: "250Mi"
+      volumes:
+      - name: config
+        configMap:
+          name: matomo-config
+
+      # TODO how do we update this for a filesystem?
+      - name: html-files
+        cephfs:
+          monitors:
+          - 192.168.0.1
+          path: /matomo
+          user: admin
+          secretRef:
+            name: logs-spack-io-secrets
+            key: logs-spack-io-secrets
+          readOnly: false
+      nodeSelector:
+        "beta.kubernetes.io/instance-type": "t2.2xlarge"
+
+
+---
```

Comment:
Let's split off each of the distinct k8s objects to their own files. (`k8s/logs-spack-io/configmaps.yaml`)